Files/ViewModels/MainPageViewModel.cs

```diff
@@ -103,15 +103,31 @@ private void NavigateToNumberedTabKeyboardAccelerator(KeyboardAcceleratorInvoked
                     break;
 
                 case VirtualKey.Tab:
-                    // Select the next tab
-                    if ((App.MainViewModel.TabStripSelectedIndex + 1) < AppInstances.Count)
+                    bool shift = e.KeyboardAccelerator.Modifiers.HasFlag(VirtualKeyModifiers.Shift);
+
+                    if (!shift) // ctrl + tab, select next tab
                     {
-                        indexToSelect = App.MainViewModel.TabStripSelectedIndex + 1;
+                        if ((App.MainViewModel.TabStripSelectedIndex + 1) < AppInstances.Count)
+                        {
+                            indexToSelect = App.MainViewModel.TabStripSelectedIndex + 1;
+                        }
+                        else
+                        {
+                            indexToSelect = 0;
+                        }
                     }
-                    else
+                    else // ctrl + shift + tab, select previous tab
```

Comment:
Instead of using an if statement, can you add a separate case to the switch statement?

Reply:
Do you mean to create a new switch statement inside the case of Tab?

Like:

case VirtualKey.Tab:
     switch (e.KeyboardAccelerator.Modifiers)

Because the key is the same on both cases, only change modificator, as happen on Ctrl + T scenario:

https://github.com/files-community/Files/blob/f503abcaea35f8c121e1aa3cf90406850c80401b/Files/Views/MainPage.xaml#L176
https://github.com/files-community/Files/blob/f503abcaea35f8c121e1aa3cf90406850c80401b/Files/Views/MainPage.xaml#L183

Reply:
You know what, the way you did it should be fine.